crates/dekaf/tests/snapshots/field_extraction_tests__number_or_string_format_number.snap

```diff
@@ -0,0 +1,176 @@
+---
+source: crates/dekaf/tests/field_extraction_tests.rs
+expression: "roundtrip(fixture_path.clone(), raw_input.as_bytes()).await?"
+---
+[
+    Ok(
+        Record(
+            [
+                (
+                    "key",
+                    String(
+                        "1234",
+                    ),
+                ),
+                (
+                    "flow_published_at",
+                    Union(
+                        0,
+                        Record(
+                            [
+                                (
+                                    "json",
+                                    String(
+                                        "null",
+                                    ),
+                                ),
+                            ],
+                        ),
+                    ),
+                ),
+                (
+                    "my_weird_field",
+                    String(
+                        "1",
+                    ),
+                ),
+            ],
+        ),
+    ),
+    Ok(
+        Record(
+            [
+                (
+                    "key",
+                    String(
+                        "1234",
+                    ),
+                ),
+                (
+                    "flow_published_at",
+                    Union(
+                        0,
+                        Record(
+                            [
+                                (
+                                    "json",
+                                    String(
+                                        "null",
+                                    ),
+                                ),
+                            ],
+                        ),
+                    ),
+                ),
+                (
+                    "my_weird_field",
+                    String(
+                        "1",
+                    ),
+                ),
+            ],
+        ),
+    ),
+    Ok(
+        Record(
+            [
+                (
+                    "key",
+                    String(
+                        "1234",
+                    ),
+                ),
+                (
+                    "flow_published_at",
+                    Union(
+                        0,
+                        Record(
+                            [
+                                (
+                                    "json",
+                                    String(
+                                        "null",
+                                    ),
+                                ),
+                            ],
+                        ),
+                    ),
+                ),
+                (
+                    "my_weird_field",
```

Comment:
this is indeed a weird field but I think this could use a better name to make it easier to follow the snapshots 😁 
perhaps something signifying what the field is actually... `string_int_fmt_number`?